go/service/main.go

```diff
@@ -89,16 +92,19 @@ func (d *Service) Handle(c net.Conn) {
 	server.AddCloseListener(cl)
 	connID := d.G().NotifyRouter.AddConnection(xp, cl)
 
-	if err := d.RegisterProtocols(server, xp, connID, d.G()); err != nil {
+	// create one log queue per connection
+	logq := newLogQueue()
+
+	if err := d.RegisterProtocols(server, xp, connID, logq, d.G()); err != nil {
 		d.G().Log.Warning("RegisterProtocols error: %s", err)
 		return
 	}
 
 	if d.isDaemon {
-		baseHandler := NewBaseHandler(xp)
-		logUI := LogUI{sessionID: 0, cli: baseHandler.getLogUICli()}
-		handle := d.G().Log.AddExternalLogger(&logUI)
-		defer d.G().Log.RemoveExternalLogger(handle)
+		// if this is the daemon, then add the log queue to the forwarder
+		d.logForwarder.Add(logq)
```

Comment:
So this will happen for electron apps too, right?  All clients have a log queue, even if they're never used?